src/hooks/useGetTokenListToQuery.tsx

```diff
@@ -0,0 +1,104 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+
+interface TokenListState {
+  ethereum?: [];
+  polygon?: [];
+  avalanche?: [];
+  fantom?: [];
+  binance?: [];
+  arbitrum?: [];
+}
+
+interface IChainIdNameMap {
+  1: string;
+  137: string;
+  56: string;
+  42161: string;
+  43114: string;
+  250: string;
+}
+export function useGetTokenListToQuery() {
+  const [tokenList, setTokenList] = useState<TokenListState>();
+
+  const chainIdNameMap = {
+    1: "ethereum",
+    137: "polygon",
+    56: "binance",
+    42161: "arbitrum",
+    43114: "avalanche",
+    250: "fantom",
+  };
+
+  useEffect(() => {
+    // axios call here to get tokens
+    // const fetchTokens = async () => {
+    //   // token source could be aggregated lists or maybe goplus?
+    //   const tokenSource = "https://tokens.coingecko.com/uniswap/all.json";
+    //   const res = await axios.get(tokenSource);
+    //   const tokenSourcePolygon =
+    //     "https://api-polygon-tokens.polygon.technology/tokenlists/allTokens.tokenlist.json";
+    //   const resPolygon = await axios.get(tokenSourcePolygon);
+    //   // set tokens in state
+    //   setTokenList({
+    //     ethereum: res?.data.tokens,
+    //     polygon: resPolygon?.data.tokens,
+    //   });
+    // };
+
+    // fetchTokens();
+
+    const fetchTokensDemo = async () => {
+      // token source could be aggregated lists or maybe goplus?
```

Comment:
I think we're probably going for Go+
It's not scalable to run archival nodes for all the chains.